index/scorch/introducer.go

```diff
@@ -209,19 +215,21 @@ func (s *Scorch) introduceMerge(nextMerge *segmentMerge) {
 	for i := range s.root.segment {
 		segmentID := s.root.segment[i].id
 		if segSnapAtMerge, ok := nextMerge.old[segmentID]; ok {
-			// this segment is going away, see if anything else was deleted since we started the merge
-			if s.root.segment[i].deleted != nil {
-				// assume all these deletes are new
-				deletedSince := s.root.segment[i].deleted
-				// if we already knew about some of them, remove
-				if segSnapAtMerge.deleted != nil {
-					deletedSince = roaring.AndNot(s.root.segment[i].deleted, segSnapAtMerge.deleted)
-				}
-				deletedSinceItr := deletedSince.Iterator()
-				for deletedSinceItr.HasNext() {
-					oldDocNum := deletedSinceItr.Next()
-					newDocNum := nextMerge.oldNewDocNums[segmentID][oldDocNum]
-					newSegmentDeleted.Add(uint32(newDocNum))
+			if segSnapAtMerge != nil {
```

Comment:
maybe fold this into the previous line so there's some less "indentation" of deep blocks, like...

    if segSnapAtMerge, ok := ...; ok && segSnapAtMerge != nil {

Reply:
one reason for doing it this way was to avoid the `else` part of the if check.
if segSnapAtMerge, ok := ...; ok { 
}. Keeping it this way let us skip the else part of appending to the new snapshot created by getting into the `if` part and not doing anything. 
not sure whether there is a cleaner way to do this..Or I may be missing something...?

Reply:
i see -- i read the nesting of if-else incorrectly -- you're right.

Reply:
instead, maybe smush the if-checks on line 218 and 220?

not a big thing, but deeper indentation or nesting of the IF blocks increases some cognitive load